R/pdf_array_job.R

```diff
@@ -0,0 +1,95 @@
+#' Submit an array job to plot PDFs, then bind back together.
+#'
+#' @param jobs_file .csv that has one row for each job you want to run, with arguments for plots as columns.
+#' @param r_script Full path to the R script that has your plotting code.
+#' @param final_out_dir Path to directory to save final file. Needs to be to a path on the cluster. 
+#' @param temp_out_dir Temporary path to save files in. Defaults to /ihme/scratch/projects/covid/pdf_temp/USERNAME.
+#' @param write_file_name Final file name to write. Defaults to "all_jobs.pdf". 
+#' @param rshell R-shell to use. Defaults to LBD shell. 
+#' @param fthread fthreads for cluster jobs. Defaults to 5. 
+#' @param fmem fmem for cluster jobs. Defaults to 10G. 
+#' @param h_rt h_rt for cluster jobs. Defaults to 2 hours. 
+#' @param queue Queue for cluster jobs. Options are 'all.q', 'i.q', 'long.q', or 'd.q'. Defaults to 'all.q'. 
+#' @param project Cluster project. Defaults to 'proj_covid'. 
+#' @param job_name Cluster job name. Defaults to 'pdf_array'. 
+#' @param output Path to write cluster output messages to. Defaults to "/share/temp/sgeoutput/USERNAME/output"
+#' @param errors Path to write cluster error messages to. Defaults to "/share/temp/sgeoutput/USERNAME/errors"
+#' @param remove_temp_files Remove temporary files after jobs have completed? Defaults to TRUE.
+pdf_array_job <- function(jobs_file, r_script, final_out_dir, 
+                          temp_out_dir=NULL, write_file_name="all_jobs.pdf",
+                          rshell="/share/singularity-images/lbd/shells/singR.sh -e s", 
+                          fthread=5, fmem='10G', h_rt="00:02:00:00", queue="all.q",
+                          project="proj_covid", job_name="pdf_array", output=NULL, errors=NULL, 
+                          remove_temp_files=TRUE){
+  require(data.table) 
```

Comment:
`data.table` is already in the "Imports" part of `DESCRIPTION`. If you just namespace the data.table calls this an be safely removed.